### PR TITLE
Abort on `std::logic_error` in CI

### PR DESCRIPTION
**Pull request by tavplubix**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

Usually `std::logic_error` indicates a bug, just like our `LOGICAL_ERROR`. 
See also https://github.com/ClickHouse/ClickHouse/pull/51135

### CODE REVIEW

**Comment by robot-clickhouse-ci-2** [comment]
This is an automated comment for commit 6cbcfa9a698b6897e355538cd7683c003fad4056 with description of existing statuses. It's updated for the latest CI running
The full report is available [here](https://s3.amazonaws.com/clickhouse-test-reports/51907/6cbcfa9a698b6897e355538cd7683c003fad4056/ci_running.html)
The overall status of the commit is 🟡 pending

<table><thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody><tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td>🟡 pending</td></tr>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td>🟢 success</td></tr>
</table>

**Comment by tavplubix** [comment]
Integration tests (asan, analyzer) [1/6] - `test_parallel_replicas_skip_shards` was broken in master
Integration tests (release) [3/4]  - `test_storage_azure_blob_storage/test.py::test_schema_inference_cache` is flaky
Upgrade check (debug) - https://github.com/ClickHouse/ClickHouse/issues/51586

File: src/Common/Exception.cpp [hunk 1]
```diff
@@ -429,17 +429,20 @@ PreformattedMessage getCurrentExceptionMessageAndPattern(bool with_stacktrace, b
...
-//         catch (...) {}
-// #endif
+#ifdef ABORT_ON_LOGICAL_ERROR
+        try
+        {
+            throw;
+        }
+        catch (const std::logic_error &)
+        {
+            if (!with_stacktrace)
```
Comment (antonio2368) [inline src/Common/Exception.cpp:439]: Did you mean 
```
if (with_stacktrace)
```

Comment (tavplubix) [reply src/Common/Exception.cpp:439]: No, if with_stacktrace is true, then we already have the stacktrace printed to `stream`. And we need to print it if it's false

**Pull request merged**